maskrcnn_benchmark/engine/inference.py

```diff
@@ -386,6 +385,35 @@ def inference(
     if output_folder:
         torch.save(predictions, os.path.join(output_folder, "predictions.pth"))
 
+    if 'coco' in dataset_name:
+        results, coco_results = do_coco_evaluation(predictions=predictions,
+                                                   dataset=dataset,
+                                                   box_only=box_only,
+                                                   output_folder=output_folder,
+                                                   iou_types=iou_types,
+                                                   logger=logger,
+                                                   expected_results=expected_results,
+                                                   expected_results_sigma_tol=expected_results_sigma_tol)
+    elif 'voc' in dataset_name:
+        results = do_voc_evaluation(predictions=predictions,
+                                    dataset=dataset,
+                                    output_folder=output_folder,
+                                    logger=logger)
+        coco_results = None
+    else:
+        raise NotImplementedError('Unsupported dataset type %s.' % dataset_name)
+
+    return results, coco_results, predictions
+
+
+def do_coco_evaluation(predictions,
```

Comment:
What do you think about moving the coco-specific and pascal-specific evaluation functions into their own file, and have the inference file here be very agnostic to the dataset? It is almost there, right?